e2e-testing.md

```diff
@@ -0,0 +1,321 @@
+# E2E Testing Style Guide <!-- omit in toc -->
+
+[go/e2e-testing-style-guide](http://go/e2e-testing-style-guide)
+
+- [1\. Factory classes](#1-fixtures)
+    - [1.1. Use Factory classes only for Entities](#11-use-factory-classes-for-entities)
+- [2\. Page objects](#2-page-objects)
+    - [2.1. Define page locators outside the constructor](#21-define-page-locators-outside-the-constructor)
+    - [2.2. Parametrized page URLs defining](#22-parametrized-page-urls-defining)
+    - [2.3. Use Components to re-use common page element groups](#23-use-components-to-re-use-common-page-element-groups)
+    - [2.4. Do not use dynamic values in allure step text](#24-do-not-use-dynamic-values-in-allure-step-text)
+- [3\. Test scenarios](#2-test-scenarios)
+  - [3.1. Use snake_case for test folders naming](#31-use-snake-case-for-test-folders-naming)
+  - [3.2. Use folders for test suite organization](#32-use-folders-for-test-suite-organization)
+  - [3.3. Use one spec file per one test](#33-use-one-spec-per-one-test)
+  - [3.4. Before block should not contain common test data preparation](#34-before-block-should-not-contain-common-test-data-preparation)
+  - [3.5. Test block should only contain test steps_and_assertions](#35-test-block-should-only-contain-test-steps-and-assertions)
+- [4\. Assertions](#2-assertions)
+  - [4.1 One assertion per one test step method](#41-one-assertion-per-one-test-step-method)
+  - [4.2 Use expect for assertions](#42-use-expect-for-assertions)
+
+
+1\. Factory classes
+--------------
+
+#### 1.1. Use Factory classes only for Entities
+
+The ```FactoryItem``` class should be extended for each new Entity like User, Course, Event, StudentGroup class.
```

Comment:
Sometimes, "relation" entities might be complicated and require their own Factory. Like TestTaskAttempt or even CourseUser.
I agree with keeping this rule, just pointing to edge cases (without a solution🙃).